src/test/assembly/asm/hexagon-hvx64.rs

```diff
@@ -0,0 +1,80 @@
+// no-system-llvm
+// assembly-output: emit-asm
+// compile-flags: --target hexagon-unknown-linux-musl
+// compile-flags: -C target-feature=+hvx
+// compile-flags: -C target-feature=+hvx-length64b
+
+#![feature(no_core, lang_items, rustc_attrs, repr_simd)]
+#![crate_type = "rlib"]
+#![no_core]
+#![allow(asm_sub_register, non_camel_case_types)]
+
+#[rustc_builtin_macro]
+macro_rules! asm {
+    () => {};
+}
+#[rustc_builtin_macro]
+macro_rules! stringify {
+    () => {};
+}
+
+#[lang = "sized"]
+trait Sized {}
+#[lang = "copy"]
+trait Copy {}
+
+type ptr = *mut u8;
+
+#[repr(simd)]
+pub struct i8x64(i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8);
+#[repr(simd)]
+pub struct i16x32(i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16, i16);
+#[repr(simd)]
+pub struct i32x16(i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32, i32);
+
+impl Copy for i8 {}
+impl Copy for i16 {}
+impl Copy for i32 {}
+impl Copy for *const i32 {}
+impl Copy for i8x64 {}
+impl Copy for i16x32 {}
+impl Copy for i32x16 {}
+extern "C" {
+    fn extern_func();
+    static extern_static: u8;
+}
+
+macro_rules! check_transfer {
+    ($func:ident $ty:ident $class:ident) => {
+        #[no_mangle]
+        pub unsafe fn $func(x: $ty) -> $ty {
+            // Hack to avoid function merging
+            extern "Rust" {
+                fn dont_merge(s: &str);
+            }
+            dont_merge(stringify!($func));
+
+            let y;
+            asm!("{} = {}", out($class) y, in($class) x);
+            y
+        }
+    };
+}
+
+// CHECK-LABEL: vec_32x16:
+// CHECK: InlineAsm Start
+// CHECK: v{{[0-9]+}} = v{{[0-9]+}}
+// CHECK: InlineAsm End
+check_transfer!(vec_32x16 i32x16 vreg);
+
+// CHECK-LABEL: vec_16x32:
+// CHECK: InlineAsm Start
+// CHECK: v{{[0-9]+}} = v{{[0-9]+}}
+// CHECK: InlineAsm End
+check_transfer!(vec_16x32 i16x32 vreg);
+
+// CHECK-LABEL: vec_8x64:
+// CHECK: InlineAsm Start
+// CHECK: v{{[0-9]+}} = v{{[0-9]+}}
+// CHECK: InlineAsm End
+check_transfer!(vec_8x64 i8x64 vreg);
```

Comment:
These are failing with `'hvx-length64b' target feature is not enabled` -- it's not yet obvious to me why.  I'll keep digging.

Reply:
You need to define the hexagon target features in src/librustc_codegen_llvm/llvm_util.rs.

Reply:
Ok, thanks again.